### DISCUSSION TITLE
Using git-lfs as "file-backend"

### DISCUSSION THREAD

**Discussion by ZuoyunZheng** [treeverse/dvc#10444] [Help] [answered]
Hi all,
i have an on-premise git-lfs repository that contains my data that i want to use as my remote storage. I want to use dvc to be able to checkout my data according to the commit hash of my data repository. As far as i can see from the new [blog ](https://dvc.ai/blog/dvc-git-lfs) about git-lfs support, there is no mentioning of whether it is possible to use `dvc push` to also push altered versions of data to my data repo. At this point, is using git submodule easier? Is there any way to do this in the frame of dvc?

Don't know if i'm missing something so any help is appreciated. Thanks in advance!

**Comment by dberenbaum** [comment]
Hi @ZuoyunZheng, do you mean you want to push data back to a data repo while working outside of that repo or commit? If so, this is not something that is supported now with or without git-lfs. You need to checkout the data repo to push to it.

**Reply by ZuoyunZheng** [reply]
Thanks for the response.

Yes. Basically, i want to use `dvc checkout v1` to basically do something like `git lfs checkout v1` and then `dvc push v2` to `git add -a; git commit -m "v2"; git lfs push`. Is it at all possible to customize this kind of behavior by i.e. writing hooks for myself?

**Reply by dberenbaum** [reply, marked as answer]
Can you clarify whether the data is inside the same repo you are working in?

If not, then I think you may want to take a look at the discussion in https://github.com/iterative/dvc/issues/8103. It is not specific to git lfs but more about how to upload data changes back to an external "data registry" repo.

**Comment by ZuoyunZheng** [comment]
No the data are indeed **not** in the same repo i'm working in.

Thanks for the pointer. I think data registry made it click for me. What i wanted was to use git-lfs to store my data since i would like to have a clear and plain overview of the files and their history (as opposed to having to parse through all the md5 values from dvc). But that's redundant and defeating the whole purpose of dvc now that i am using dvc to track my data.
